lib/libutee/include/apdu_pta.h

```diff
@@ -0,0 +1,30 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/*
+ * Copyright (C) 2021, Foundries Limited
+ */
+
+#ifndef __APDU_PTA_H
+#define __APDU_PTA_H
+
+#define PTA_APDU_UUID { 0x3f3eb880, 0x3639, 0x11ec, \
+			{ 0x9b, 0x9d, 0x0f, 0x3f, 0xc9, 0x46, 0x8f, 0x50 } }
+/*
+ * ADPU based communication with the SE05X secure element
```

Comment:
I am writing this code as part of a more extensive integration with a service daemon. header, and buffers are generic but the request type might be only for the SE050 (although other implementations might want to use these params as well so the TXRX functions can be simplified)

This is how the types get handled on the Plug-and-Trust stack:
https://github.com/foundriesio/plug-and-trust/commit/5caf02c122f5fbb77aa41592d4b9ca0c6ef2c188#diff-4271878f3b672bd62c729c47fceec849f31ee69cc403df460d71bccd36e6ce7fR10

Reply:
My point here is to question whether this PTA exposes generic commands (in which case "APDU PTA" sounds ok), or if it the PTA API is dedicated to SE050 APDU support. Maybe the PTA expects to be generic but it must expose dedicated commands for SE050 support since some extra config info (as `SE050_APDU_TYPE_*` IDs are).

Reply:
There is nothing specific to the SE50 in this software API. Communication via APDU frames consist on sending and receiving buffers. And I added an extra param to help with the handling of those buffers in whichever way the driver considers necessary.

But if you prefer I can rename it all and make it SE050 specific - just to save time (upstreaming is expensive as everyone knows). And then we can come up with a different API if ever any other secure element gets merged in op-tee....shall I do that?

Reply:
AFAICT generic APDU does not consider types info carried by `param[0].value.a` and this makes the API not generic. Could there be a way to have a generic PTA for APDU? Is there a need?
All in one, i think its not really clean to have an APDU PTA which interface only applies to SE05x devices.

This comment applies also to https://github.com/OP-TEE/optee_client/pull/290. Is libapduteec designed only for OP-TEE relying on a SE05x device?

I apology, I don't know anything about APDUs. Maybe someone can give some feedback on that concern.

Reply:
But I disagree with that. The API is generic: ie it provides an interface to send and receive a buffer plus an enum to characterize - if necessary - the processing of those buffers.  
The SE050 processing benefits from having that enum but others might not (or might too). I guess I dont see why that makes it non-generic and not the opossite.
could you clarify? please

WRT to OP-TEE/optee_client#290 is just the same thing. Just as this PR is not SE05x specific.

Reply:
Maybe I'm the only one wondering about if this API allows to interface other devices. Should they use TYPE_NONE if not relevant? Client of the libapduteec is expected to know when it interfaces a SE05x and  set a non-TYPE_NONE?

If I'm the only one wondering about that, I guess I misunderstand the context. If so, fine with the interface but remove SE05x reference from the PTA API header file. Or change command name to highlight it's an APDU typed command that complies with SE05x (maybe other devices?) ADPU types definition.